src/remarx/app/utils.py

```diff
@@ -127,6 +137,57 @@ def create_temp_input(
         pathlib.Path.unlink(temp_file.name)
 
 
+def summarize_corpus_selection(
+    selection: _HasPath | str | pathlib.Path | None,
+) -> dict[str, int | str] | None:
+    """
+    Summarize a sentence corpus CSV selected in the UI.
+
+    Accepts either a file-browser selection (with a ``path`` attribute) or a
+    filesystem path and returns corpus statistics that can be displayed in the
+    Marimo table.
+    """
+
+    path_value = getattr(selection, "path", selection)
+    if path_value is None:
+        return None
+
+    path = pathlib.Path(path_value)
+    if not path.is_file():
+        return None
+
+    # Use lazy scanning so even large corpora only require lightweight
+    # aggregations while collecting summary statistics.
+    corpus = pl.scan_csv(path, infer_schema_length=0)
+    total_sentences = corpus.select(pl.len()).collect().item()
+
+    if "section_type" in corpus.collect_schema():
+        counts = (
+            corpus.group_by("section_type").agg(pl.len().alias("sentences")).collect()
+        )
+        body_sentences = int(
+            counts.filter(pl.col("section_type") == "text")["sentences"].sum()
+        )
+        footnote_sentences = int(
+            counts.filter(pl.col("section_type") == "footnote")["sentences"].sum()
+        )
+    else:
+        body_sentences = total_sentences
+        footnote_sentences = 0
+
+    last_updated = datetime.fromtimestamp(path.stat().st_mtime).strftime(
+        "%Y-%m-%d %H:%M:%S"
+    )
+
+    return {
+        "corpus": path.name,
+        "total lines": int(total_sentences),
+        "body text lines": body_sentences,
+        "footnote lines": footnote_sentences,
```

Comment:
Revise labels to more precise & consistent. 
```suggestion
        "filename": path.name,
        "total sentences": int(total_sentences),
        "body text sentences": body_sentences,
        "footnote sentences": footnote_sentences,
```